src/irr/core/CFixedCapacityDoublyLinkedList.h

```diff
@@ -0,0 +1,117 @@
+#ifndef __DOUBLY_LINKED_LIST_H_INCLUDED__
+#define __DOUBLY_LINKED_LIST_H_INCLUDED__
+#include "irr/core/alloc/PoolAddressAllocator.h"
+#include "irr/core/Types.h"
+
+namespace irr {
+	namespace core {
+
+		template<typename Value>
+		struct Snode
+		{
+			Value data;
+			Snode* prev;
+			Snode* next;
+
+			Snode(const Value& val) : data(val)
+			{ }
+			Snode(Snode&& other) : data(std::move(other.data)), prev(std::move(other.prev)), next(std::move(other.next))
+			{ }
+
+			Snode& operator=(const Snode& other)
+			{
+				this.data = other.data;
+				this.prev = other.prev;
+				this.next = other.next;
+				return *this;
+			}
+
+			Snode& operator=(Snode&& other)
+			{
+				this.data = std::move(other.data);
+				this.prev = std::move(other.prev);
+				this.next = std::move(other.next);
+				return *this;
+			}
+		};
+
+		template<typename Value>
+		class DoublyLinkedList
+		{
+			typedef Snode<Value>* iteratorptr_t;
+
+		private:
+			void* reservedSpace;
+			PoolAddressAllocator<uint32_t> alloc;
+			iteratorptr_t p_begin;
+			iteratorptr_t p_end;
+			uint32_t cap;
+
+			inline void popBack()
+			{
+				if(p_end->prev != nullptr)
+					p_end->prev->next = nullptr;
```

Comment:
instead of `nullptr` define an `invalid_iterator`

Reply:
also `p_end` doesn't and shouldn't exist, you should keep `p_back` which is your `p_end->prev`

Reply:
usually in STL compatible containers "end" means one past the container, where as "back" is the true last element.. see std::vector and std::array

Reply:
it seems like your `p_end` functions like a `p_back`, if so, rename it.